File: service/src/main/scala/pregnaware/user/entities/WrappedUser.scala
package pregnaware.user.entities

import java.time.{Instant, LocalDate}

import pregnaware.naming.entities.WrappedBabyName

case class WrappedUser(
  userId: Int,
  displayName: String,
  email: String,
  dueDate: Option[LocalDate],
  joinedDate: LocalDate,
  lastAccessedTime: Instant,
  babyNames: Seq[WrappedBabyName],
  passwordHash: String,
  friends: Seq[WrappedFriend],
  friendRequestsSent: Seq[WrappedFriend],
  friendRequestsReceived: Seq[WrappedFriend])

File: frontend/www/app/scripts/models/wrapped-user.ts
module models {
    export interface WrappedUser {
        userId: number
        displayName: string
        email: string
        dueDate: LocalDate
        joinedDate: LocalDate
        lastAccessedTime: number,
        babyNames: WrappedBabyName[]
        friends: WrappedFriend[]
        friendRequestsSent: WrappedFriend[]
        friendRequestsReceived: WrappedFriend[]
    }
}